js/runner_test.go

```diff
@@ -268,6 +268,161 @@ func TestSetupTeardown(t *testing.T) {
 	}
 }
 
+func TestSetupDataIsolation(t *testing.T) {
+	r1, err := New(&lib.SourceData{
+		Filename: "/script.js",
+		Data: []byte(`
+			export let options = {
+				setupTimeout: "10s",
+				teardownTimeout: "10s",
+			};
+
+			export function setup() {
+				return { v: 1 };
+			}
+			export function teardown(data) {
+				if (data.v != 1) {
+					throw new Error("teardown: wrong data: " + data.v)
+				}
+				data.v = 2
+			}
+			export default function(data) {
+				if (data.v != 1) {
+					throw new Error("default: wrong data: " + JSON.stringify(data))
+				}
+				data.v = 2
+			}
+		`),
+	}, afero.NewMemMapFs(), lib.RuntimeOptions{})
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	r2, err := NewFromArchive(r1.MakeArchive(), lib.RuntimeOptions{})
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	testdata := map[string]*Runner{"Source": r1, "Archive": r2}
+	for name, r := range testdata {
+		samples := make(chan stats.SampleContainer, 100)
+		t.Run(name, func(t *testing.T) {
+			if !assert.NoError(t, r.Setup(context.Background(), samples)) {
+				return
+			}
+
+			vu, err := r.NewVU(samples)
+			if assert.NoError(t, err) {
+				for i := 0; i < 10; i++ {
+					err := vu.RunOnce(context.Background())
+					assert.NoError(t, err)
+				}
+			}
+
+			assert.NoError(t, r.Teardown(context.Background(), samples))
+		})
+	}
+}
+
+func testSetupDataHelper(t *testing.T, src *lib.SourceData) {
+	expScriptOptions := lib.Options{SetupTimeout: types.NullDurationFrom(1 * time.Second)}
+	r1, err := New(src, afero.NewMemMapFs(), lib.RuntimeOptions{Env: map[string]string{"expectedSetupTimeout": "1s"}})
+	require.NoError(t, err)
+	require.Equal(t, expScriptOptions, r1.GetOptions())
+
+	r2, err := NewFromArchive(r1.MakeArchive(), lib.RuntimeOptions{Env: map[string]string{"expectedSetupTimeout": "3s"}})
```

Comment:
Same here, `expectedSetupTimeout` seems like a copy-paste error. To be honest, I'm not sure if we actually need `r2` at all. I know that a lot of the other tests make an archive from a script like this and test both, but not sure we need to do that everywhere, the only thing it would probably gain from it is extended test execution times.